### PR TITLE
[956] Add test and maybe a fix for modules passed as arguments

### PR DESCRIPTION
**Pull request by tpitale**
Related to #956 

I want to check in early to make sure I'm on the right path.

When a module is passed as an argument to a function, it should maybe be aliased?

What I'm actually trying to fix is probably when a module is passed to a macro and not aliased. I may try to update the PR if I'm on the right path.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-erlang 24.2
+erlang 24.3.4.2
```
Comment (tpitale) [inline .tool-versions:1]: I can remove this.

File: lib/credo/check/design/alias_usage.ex [hunk 2]
```diff
@@ -61,6 +61,8 @@ defmodule Credo.Check.Design.AliasUsage do
 
   alias Credo.Code.Name
 
+  @keywords [:alias]
```
Comment (tpitale) [inline lib/credo/check/design/alias_usage.ex:64]: This could maybe be a param.

File: lib/credo/check/design/alias_usage.ex [hunk 3]
```diff
@@ -169,40 +171,90 @@ defmodule Credo.Check.Design.AliasUsage do
...
+       excluded_namespaces,
+       excluded_lastnames,
+       only,
+       aliases,
+       mod_deps
+     )}
+  end
+
+  defp find_issues(
+         {fun_atom, _, [{:__aliases__, meta, mod_list}]} = ast,
```
Comment (tpitale) [inline lib/credo/check/design/alias_usage.ex:189]: This was the best I think I could do when trying to find a function call 🤔 But, I'm not super-familiar with the AST.

**Comment by rrrene** [comment]
This is really great. Can you introduce a param to the check that allows users to deactivate this behaviour? 

Something like `ignore_plain_module_reference` (please use a better name if you have suggestions ^^)^.

**Comment by rrrene** [comment]
I am closing this for age/inactivity. 

If you find the time to make the final changes, please feel free to re-open this issue at your discretion.

**Comment by rrrene** [comment]
@tpitale I closed this in error. It is now merged. Thx for your contribution! :+1:

**Pull request merged**